c2corg_ui/static/js/documentediting.js

```diff
@@ -33,7 +33,14 @@ app.documentEditingDirective = function() {
     scope: true,
     controller: 'appDocumentEditingController',
     controllerAs: 'editCtrl',
-    bindToController: true
+    bindToController: true,
+    link: function(scope, el, attrs, ctrl) {
+      // The directive executes after the dom is loaded but we have to wait until the expressions are evaluated
```

Comment:
> not the document data, but it gives it a little bit of time for the ng-ifs. How do i know when the ng ifs are finished evaluating?

When do the `ng-if`s start evaluating? When `this.scope_['waypoint']` is set. And that is when the request has finished loading. I wouldn't try to count the elements with a class `.steps`. When you have the waypoint (when the data is loaded), I would determine how many steps you need depending on the type and then directly set `this.maxSteps_`.